zerver/lib/message.py

```diff
@@ -1765,25 +1785,37 @@ def visibility_policy_for_send_message(
     # We prioritize 'FOLLOW' over 'UNMUTE' in muted streams.
     # We need to carefully handle the following two cases:
     #
-    # 1. When a user starts a topic
-    #    note: starting a topic is INITIATION as well as PARTICIPATION
-    # 'automatically_follow_topics_policy' | 'automatically_unmute_topics_in_muted_streams_policy' | visibility_policy
-    #       ON_PARTICIPATION               |                   ON_INITIATION                       |     FOLLOWED
-    #         ON_INITIATION                |                 ON_PARTICIPATION                      |     FOLLOWED
-    #         ON_INITIATION                |                   ON_INITIATION                       |     FOLLOWED
+    # 1. When an action qualifies for multiple values. Example:
+    #    - starting a topic is INITIATION, PARTICIPATION as well as SEND
+    #    - sending a non-first message is PARTICIPATION as well as SEND
+    # action | 'automatically_follow_topics_policy' | 'automatically_unmute_topics_in_muted_streams_policy' | visibility_policy
+    #  start |    ON_PARTICIPATION / ON_SEND        |                   ON_INITIATION                       |     FOLLOWED
+    #  send  |    ON_SEND / ON_PARTICIPATION        |              ON_PARTICIPATION / ON_SEND               |     FOLLOWED
     #
-    # 2. When both the policies have the same value 'ON_PARTICIPATION'.
-    # 'automatically_follow_topics_policy' | 'automatically_unmute_topics_in_muted_streams_policy' | visibility_policy
-    #       ON_PARTICIPATION               |                 ON_PARTICIPATION                       |     FOLLOWED
+    # 2. When both the policies have the same values.
+    # action | 'automatically_follow_topics_policy' | 'automatically_unmute_topics_in_muted_streams_policy' | visibility_policy
+    #  start |         ON_INITIATION                |                   ON_INITIATION                       |     FOLLOWED
+    #  partc |       ON_PARTICIPATION               |                 ON_PARTICIPATION                      |     FOLLOWED
+    #  send  |           ON_SEND                    |                     ON_SEND                           |     FOLLOWED
     visibility_policy = None
 
     if current_visibility_policy == UserTopic.VisibilityPolicy.FOLLOWED:
         return visibility_policy
 
-    visibility_policy = visibility_policy_for_participation(sender, is_stream_muted)
+    visibility_policy_participation = visibility_policy_for_participation(sender, is_stream_muted)
+    visibility_policy_send = visibility_policy_for_send(sender, is_stream_muted)
```

Comment:
`visibility_policy_send` doesn't look a perfect name for this. Any better suggestions?

Reply:
I don't have a readily available better idea; I think it'll be clear enough from the context. So let's just live with it and we can rethink when next revising that code path.